gpytorch/metrics/metrics.py

```diff
@@ -0,0 +1,73 @@
+from math import pi
+
+import torch
+
+from ..distributions import MultivariateNormal
+
+pi = torch.tensor(pi)
+
+
+def negative_log_predictive_density(
+    pred_dist: MultivariateNormal,
+    test_y: torch.Tensor,
+):
+
+    with torch.no_grad():
+        return -pred_dist.log_prob(test_y).item()
+
+
+def mean_standardized_log_loss(
+    pred_dist: MultivariateNormal,
+    test_y: torch.Tensor,
+):
+    """
+    Mean Standardized Log Loss.
+
+    Reference: Page No. 23,
+    Gaussian Processes for Machine Learning,
+    Carl Edward Rasmussen and Christopher K. I. Williams,
+    The MIT Press, 2006. ISBN 0-262-18253-X
+    """
+
+    with torch.no_grad():
+        f_mean = pred_dist.mean
+        f_var = pred_dist.variance
+        return 0.5 * (torch.log(2 * pi * f_var) + torch.square(test_y - f_mean) / (2 * f_var)).mean().item()
+
+
+def percentile_coverage_error(
+    pred_dist: MultivariateNormal,
+    test_y: torch.Tensor,
+    percentile: float = 95,
+):
+    """
+    Percentile Coverage error.
+
+    """
+
+    assert 0 <= percentile <= 100, "Percentile must be between 0 and 100"
+
+    standard_normal = torch.distributions.Normal(loc=0.0, scale=1.0)
+    with torch.no_grad():
+        deviation = standard_normal.icdf(torch.tensor(0.5 + 0.5 * (percentile / 100)))
+        lower = pred_dist.mean - deviation * pred_dist.stddev
+        upper = pred_dist.mean + deviation * pred_dist.stddev
+        n_samples_within_bounds = ((test_y > lower) * (test_y < upper)).sum()
```

Comment:
sum over the batch dimensions (e.g. sum(-1)) here so that this supports the batched case.

Reply:
If I understand this right, in the case where test_y is `n×d`, we want the output of this metric `1×d`? Currently, it outputs a scaler as torch sums over all the dimensions by default.

Reply:
Yes, that's exactly the goal here.

Reply:
It often makes more sense to consider the metrics for each output separately.

Reply:
Yeah, makes perfect sense! Added the batch support.